### PR TITLE
Update API functions docstrings

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
| https://github.com/wazuh/wazuh/issues/11222 |

## Description

This PR is part of https://github.com/wazuh/wazuh/issues/11222.

In this pull request, I have updated the API docstrings.

### CODE REVIEW

File: api/api/api_exception.py [hunk 1]
```diff
@@ -40,6 +44,13 @@ def __init__(self, code: int, details: str = None):
         }
 
     def __str__(self):
```
Comment (vicferpoy) [inline api/api/api_exception.py:46]:
```suggestion
    def __str__(self) -> str:
```

File: api/api/authentication.py [hunk 2]
```diff
@@ -129,27 +141,35 @@ def change_keypair():
...
     ----------
     user_id : str
-        Unique username
+        Unique username.
     data : dict
-        Roles permissions for the user
+        Roles permissions for the user.
     run_as : bool
-        Indicate if the user has logged in with run_as or not
+        Indicate if the user has logged in with run_as or not.
```
Comment (vicferpoy) [inline api/api/authentication.py:167]:
```suggestion
        Indicate if the user has logged in with authorization context or not.
```

File: api/api/authentication.py [hunk 3]
```diff
@@ -175,23 +195,24 @@ def generate_token(user_id=None, data=None, run_as=False):
...
+        Unique username.
     roles : tuple
-        Tuple of roles related with the current token
+        Tuple of roles related with the current token.
     token_nbf_time : int
-        Issued at time of the current token
+        Issued at time of the current token.
     run_as : bool
-        Indicate if the token has been granted through run_as endpoint
+        Indicate if the token has been granted through run_as endpoint.
```
Comment (vicferpoy) [inline api/api/authentication.py:210]:
```suggestion
        Indicate if the token has been granted through authorization context endpoint.
```

File: api/api/controllers/cdb_list_controller.py [hunk 4]
```diff
@@ -74,11 +77,12 @@ async def get_lists(request, pretty: bool = False, wait_for_complete: bool = Fal
 
 
 async def get_file(request, pretty: bool = False, wait_for_complete: bool = False, filename: str = None,
-                   raw: bool = False):
-    """Get content of one CDB list file, in raw or dict format.
+                   raw: bool = False) -> Union[web.Response, ConnexionResponse]:
+    """"Get content of one CDB list file, in raw or dict format.
```
Comment (vicferpoy) [inline api/api/controllers/cdb_list_controller.py:81]:
```suggestion
    """Get content of one CDB list file, in raw or dict format.
```

File: api/api/controllers/default_controller.py [hunk 5]
```diff
@@ -18,12 +18,20 @@
 logger = logging.getLogger('wazuh-api')
 
 
-async def default_info(pretty=False):
-    """Get basicinfo
+async def default_info(pretty: bool = False) -> web.Response:
+    """Get basicinfo.
```
Comment (vicferpoy) [inline api/api/controllers/default_controller.py:22]: You could get rid of this line.

File: api/api/controllers/default_controller.py [hunk 6]
```diff
@@ -18,12 +18,20 @@
...
 
-    :param pretty: Show results in human-readable format
+    Returns basic information about the Wazuh API.
```
Comment (vicferpoy) [inline api/api/controllers/default_controller.py:24]:
```suggestion
    Return basic information about the Wazuh API.
```

File: api/api/util.py [hunk 7]
```diff
@@ -23,13 +36,20 @@ def serialize(item):
...
-def _deserialize(data, klass):
-    """Deserializes dict, list, str into an object.
+def _deserialize(data: Union[dict, list, str], klass: type) -> object:
+    """Deserialize dict, list, str into an object.
 
-    :param data: dict, list or str.
-    :param klass: class literal, or string of class name.
+    Parameters
+    ----------
+    data : Union[dict, list, str]
```
Comment (vicferpoy) [inline api/api/util.py:44]: This syntax is not correct. In the docstrings, you use `or`.

**Review by vicferpoy — CHANGES_REQUESTED** [review]

File: api/api/util.py (hunk 7 above)
Comment (mcarmona99) [reply api/api/util.py:44]: I have also reviewed all the parameters using the `Union` keyword.

**Pull request merged**